packages/legend-query-builder/src/components/QueryBuilderConstantExpressionPanel.tsx

```diff
@@ -105,6 +114,12 @@ const QueryBuilderConstantExpressionEditor = observer(
       constantState.setValueSpec(valSpec);
     };
 
+    const saveConstantName = (): void => {
+      if (!validationMessage) {
+        variableExpression_setName(varExpression, constantNameValue);
+      }
```

Comment:
Do we want to show a warning to the user after they exit the constant panel that we haven't saved the constant name they have updated?

Reply:
That's true, might be good safeguard. I added now an update where if user navigates away and its invalid, then we will have a caution action alert that lets them know 👍